aztk/client/base/helpers/get_application_log.py

```diff
@@ -68,7 +68,9 @@ def get_log(batch_client, blob_client, cluster_id: str, application_name: str, t
 
     task = __wait_for_app_to_be_running(batch_client, cluster_id, application_name)
 
-    if not __check_task_node_exist(batch_client, cluster_id, task):
+    #TODO: find a better way to detect ghost tasks -- metatdata
```

Comment:
This needs to be done before merging. Only detecting `TaskState.completed` is too flimsy, especially around many concurrent application submits.